packages/mobx-react-lite/__tests__/observer.test.tsx

```diff
@@ -270,7 +270,10 @@ function runTestSuite(mode: "observer" | "useObserver") {
 
     describe("issue 309", () => {
         test("isObserverBatched is still defined and yields true by default", () => {
+            const warn = jest.spyOn(console, "warn").mockImplementation(() => {})
             expect(isObserverBatched()).toBe(true)
+            expect(warn).toHaveBeenCalled()
+            warn.mockReset()
```

Comment:
Shouldn't it be `mockRestore`?
Also what happens if the test throws before `mockRestore`? Does each test run in complete isolation or will it after other tests and therefore we should restore console in  `try {} finally {}`/`afterEach`?

Reply:
It makes sense. I suggest to use something like `afterEach`